contents/tutorials/nuxtjs-ab-tests.md

```diff
@@ -0,0 +1,264 @@
+      ctaText.value = 'Control variant';
+    } else if (posthog.getFeatureFlag('nuxt-button-text-experiment') === 'test') {
+      ctaText.value = 'Test variant';
+    }
+  }
+});
+
+const captureEventInPostHog = () => {
+  // ...rest of code
+};
+</script>
+```
+
+Now if you refresh your app, you should see the button text updated to either `Control variant` or `Test variant`. 
+
+You can also test either variant by adding an override in `posthog.client.js`:
+
+```js file=plugins/posthog.client.js
+import { defineNuxtPlugin } from '#app'
+import posthog from 'posthog-js'
+
+export default defineNuxtPlugin(nuxtApp => {
+  // ... your existing code
+  
+  posthogClient.featureFlags.override({'nuxt-button-text-experiment': 'test'});
+  return {
+    provide: {
+      posthog: () => posthogClient
+    }
+  }
+})
+```
+
+With this, you’re ready to launch your experiment, but make sure to remove your override!
+
+### Server-side rendering
+
+Notice that when you refresh the page, the button text flickers between `No variant` and `Control/Test variant`. This is because it takes time for PostHog to load and make the feature flag request.
+
+Server-side rendering is a way to avoid this. This fetches the feature flag before the page loads on the client.
+
+To set this up, we must install and use [PostHog’s Node library](/libraries/node) (because we are making server-side requests).
+
+```bash
+npm install posthog-node
+```
+
+Then, we use the PostHog Node library to fetch the feature flag using `useAsyncData` in `app.vue`:
+
+```vue file=app.vue
+ <!-- ...rest of code -->
+
+<script setup>
+import { useAsyncData, useCookie, useRuntimeConfig } from 'nuxt/app';
+import { PostHog } from 'posthog-node';
+
+const { data: ctaData, error } = await useAsyncData('ctaText', async () => {
+  let ctaText = 'No Variant';
+  const runtimeConfig = useRuntimeConfig();
+  const posthog = new PostHog(
+    runtimeConfig.public.posthogPublicKey,
+    { host: runtimeConfig.public.posthogHost }
+  );
+
+  const cookies = useCookie(`ph_${runtimeConfig.public.posthogPublicKey}_posthog`);
+  const userId = cookies.value.distinct_id;
+  if (userId) {
+    try {
+      const enabledVariant = await posthog.getFeatureFlag('nuxt-button-text-experiment', userId);
+      if (enabledVariant === 'control') {
+        ctaText = 'Control Variant';
+      } else if (enabledVariant === 'test') {
+        ctaText = 'Test Variant'
+      }
+    } catch (error) {
+      console.log(error);
+    }
+  }
+  return ctaText;
+});
+const ctaText = computed(() => ctaData.value);
+
+const captureEventInPostHog = () => {
+  // ... rest of code
+};
+
+</script>
+```
+
+Now, when you refresh the page, the button text is already set when the page loads. 
+
+> **Note:** This only works on subsequence visits to your app where the PostHog cookie is set. Using this method still requires waiting for PostHog and feature flags to load on the first visit.
```

Comment:
The eternal struggle with these tutorials is that really they require bootstrapping but aren't bootstrapping tutorials 😅 

With this code, the first time is broken. Could you add a little more code to fall back to the client side evaluation if the ctaText isn't set? I don't think we need to do the bootstrapping detials for this one, just reuse the last section.

![CleanShot 2023-12-01 at 10 06 15@2x](https://github.com/PostHog/posthog.com/assets/34755028/87a0f9da-d115-4dbc-bb6e-58cad9fd4ca8)